invokeai/backend/stable_diffusion/diffusion/custom_atttention.py

```diff
@@ -188,11 +199,14 @@ def __call__(
         hidden_states = attn.to_out[1](hidden_states)
 
         if input_ndim == 4:
+            batch_size, channel, height, width = hidden_states.shape
             hidden_states = hidden_states.transpose(-1, -2).reshape(batch_size, channel, height, width)
 
         if attn.residual_connection:
             hidden_states = hidden_states + residual
 
         hidden_states = hidden_states / attn.rescale_output_factor
+        # End of unmodified block from AttnProcessor2_0
 
-        return hidden_states
+        # casting torch.Tensor to torch.FloatTensor to avoid type issues
+        return cast(torch.FloatTensor, hidden_states)
```

Comment:
It looks like we relaxed all of the input parameter types from`torch.FloatTensor` to `torch.Tensor`, but then added a cast here to make the inferred type more specific. I feel like we should go one way or the other (and it's fine with me if that's `torch.Tensor` everywhere). Is there something more to this that I'm missing?

Reply:
It was mainly done to avoid the strict typing checks we have. All our ops on tensors return `torch.Tensor` and not torch.FloatTensor which eventually defaults to `torch.FloatTensor` anyway. So rather than casting everything else, I thought it was just simpler to change everything to `torch.Tensor` and then just cast it to `FloatTensor` in the end to make sure the `__call__` override matches the parent function.